### ISSUE TITLE
Inquiry about Input for UTPlaceF from Vivado

### ISSUE THREAD

**Issue by zslwyuan** [zzzDavid/RapidLayout-UTPlaceF#1]
Hi!

I am recently trying to test UTPlaceF with some designs from Vivado. Have you tried to translate the files from the Vivado DCP files to the required format of UTPlaceF?

Thanks in advance! ^_^

Best regards,
Tingyuan

**Comment by zzzDavid** [comment]
Hi!

In my situation I had a systolic array design that was written in Verilog. I tried to translate it to UTPlaceF's design format in the first place but did not find any feasible method for that. So I wrote the design again to accustom UTPlaceF's need. I learned about the format from examples given at ISPD2016 [contest](http://www.ispd.cc/contests/16/ispd2016_contest.html). Specifically, I wrote a python script to generate the design files.

If you find any means that could translate verilog files into their format, I would love to know.


Thanks,

Niansong

**Comment by zslwyuan** [comment]
Hi @zzzDavid ,

Thanks a lot for your prompt reply!
In order to get rid of the tedious works, I am implementing a script-based tool to translate the content in different files. When I finish the implementation, I will share it with you! ^_^
 
Best regards,
Tingyuan

**Comment by kongqi-best** [comment]
Hi!

I'm trying to generate route congestion report by using dcp files and placement(.pl) files.  I have downloaded some benchmarks from http://www.ispd.cc/contests/17/downloads/2016/ and I want to do some place & route on them.The files contain a .dcp but when I try to open it with open_checkpoint I get this error.
Xilinx Employee explained that DCP files are for a ispd 2016 program and have been intentionally disable for opening in a standard Vivado setup.  They are only intended to be opened by participants in the ispd 2016 program.
I wonder to know if you appear error when you use the dcp files to opening checkpoint, how did you solve this problem?
(I found patch form https://fpga.socs.uoguelph.ca/benchmarks/Xilinx, does it ues for solve error? )

thanks!

Best wishes,
qikong

**Comment by zslwyuan** [comment]
> Hi!
> 
> I'm trying to generate route congestion report by using dcp files and placement(.pl) files. I have downloaded some benchmarks from http://www.ispd.cc/contests/17/downloads/2016/ and I want to do some place & route on them.The files contain a .dcp but when I try to open it with open_checkpoint I get this error.
> Xilinx Employee explained that DCP files are for a ispd 2016 program and have been intentionally disable for opening in a standard Vivado setup. They are only intended to be opened by participants in the ispd 2016 program.
> I wonder to know if you appear error when you use the dcp files to opening checkpoint, how did you solve this problem?
> (I found patch form https://fpga.socs.uoguelph.ca/benchmarks/Xilinx, does it ues for solve error? )
> 
> thanks!
> 
> Best wishes,
> qikong

Hi Qikong,

I also contacted Xilinx about this problem. Then...., I gave up the dcp files/ISPD benchmarks because they set too many constraints for our research. For long-term consideration, I determined to get away from these benchmarks...

I wrote scripts to extract information from the benchmarks and the Vivado GUI into plain text files. I have implemented an FPGA placer that can process these data and generate Tcl script to place the elements in Vivado. The FPGA placer will be available on GitHub in Sep 2021. 

I am not sure what you want to implement, but these might be clues to solve your problem.

Cheer up!

**Comment by kongqi-best** [comment]
Hi!

Thanks a lot for your prompt reply! 

My work need different placer to compare performance, previous realated works were built on this...I will try again, then decide whether to give up these benchmarks.  But I look forward to seeing the results of yours FPGA placer, and i believe it will greatly promote my work.

Thanks for your encouragement!

**Comment by zslwyuan** [comment]
> Hi!
> 
> Thanks a lot for your prompt reply!
> 
> My work need different placer to compare performance, previous realated works were built on this...I will try again, then decide whether to give up these benchmarks. But I look forward to seeing the results of your FPGA placer, and i believe it will greatly promote my work.
> 
> Thanks for your encouragement!

Hi Qikong,

OK, I see. Although our paper will be available in ICCAD 2021 (Nov, 2021) and we have also included state-of-art congestion solutions as parts of our placer (not our focus/novelty), I noticed some things that might be useful for your later implementation:

1. I would suggest you to test real-world benchmarks since ISPD benchmarks were randomly generated. Therefore, something that works for ISPD benchmarks might fail in real-world applications, which have some specific features.

2. It seems that RippleFPGA-like congestion algorithm has provided congestion estimation which is accurate enough according to our observations. It seems that fine-grained actual congestion calculation is unnecessary since congestion is usually caused by multiple sites. Therefore, maybe you don't have to get the congestion report from Vivado.

3. Timing is the most important result compared to wirelength and congestion, **for FPGA placement**, according to the comments from our reviewers.

Cheer up!
